### PR TITLE
Adding notebook checks to `cirq-superstaq`

### PR DESCRIPTION
**Pull request by pgoiporia**

### CODE REVIEW

**Comment by pgoiporia** [comment]
Opening this for review, but maybe we should hold off on merging this until we properly implement dry runs, because as it stands, every time we run notebook tests, it probably costs a little bit of money.

**Comment by vtomole** [comment]
Moving to draft based on the comment above.

**Comment by pgoiporia** [comment]
Turns out none of these notebooks use "service.run", so we can't dry run anyways. Will include `aqt.ipynb` when we have the configs sorted out. Ready for review @vtomole

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -164,3 +164,23 @@ jobs:
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[dev]
+      - name: Notebook check
+        run: |
+          check/pytest_.py --notebook
```
Comment (vtomole) [inline .github/workflows/ci.yml:186]: Remove the newline at the end of this file

File: docs/getting_started_cirq_superstaq.ipynb [hunk 2]
```diff
@@ -10,7 +10,7 @@
   },
```
Comment (vtomole) [inline docs/getting_started_cirq_superstaq.ipynb:10]: <pre class="ql-syntax" spellcheck="false">Line #2.    # print("Job status: ", job.status())
</pre><p>Why are these commented out?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/cirq-superstaq/pull/320/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Comment by vtomole** [comment]
> Turns out none of these notebooks use "service.run", so we can't dry run anyways.

True, but we want `create_job` to `dry_run` as well. As it stands, executing the `finance.ipynb` notebooks as part of the notebook check will actually submit jobs to AWS and IBM

File: docs/getting_started_cirq_superstaq.ipynb (hunk 2 above)
Comment (pgoiporia) [reply docs/getting_started_cirq_superstaq.ipynb:10]: It takes too long to run I believe
Can double check

Comment (pgoiporia) [reply docs/getting_started_cirq_superstaq.ipynb:10]: Ah ok nvm it's just a create_job.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pgoiporia) [reply .github/workflows/ci.yml:186]: I don't see a newline

Comment (vtomole) [reply .github/workflows/ci.yml:186]: Sorry, you actually need a newline. See https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

**Comment by vtomole** [comment]
What should we do with the commented out lines? Should we leave them, merge this in and create an issue to un-comment them? For example, i'm guessing the `submit_qubo` is commented cause we don't actually want to submit a qubo. This probably means that we should add `dry-run` to `submit_qubo` as well.

File: examples/finance.ipynb [hunk 3]
```diff
@@ -10,22 +10,33 @@
   },
```
Comment (vtomole) [inline examples/finance.ipynb:10]: <pre class="ql-syntax" spellcheck="false">Line #1.    # out = service.tsp(["Enter", "Cities", "Here"])
</pre><p>Why is this commented out?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/cirq-superstaq/pull/320/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (vtomole) [inline examples/finance.ipynb:10]: <pre class="ql-syntax" spellcheck="false">Line #1.    # result = service.submit_qubo(qubo, target="d-wave_dw-2000q-6_qpu", repetitions=1000)
</pre><p>Why is this commented out?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/cirq-superstaq/pull/320/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Comment by pgoiporia** [comment]
> What should we do with the commented out lines? Should we leave them, merge this in and create an issue to un-comment them? For example, i'm guessing the `submit_qubo` is commented cause we don't actually want to submit a qubo. This probably means that we should add `dry-run` to `submit_qubo` as well.

Yeah I think this makes sense

**Comment by vtomole** [comment]
https://github.com/SupertechLabs/general-superstaq/issues/174

**Comment by pgoiporia** [comment]
Deleted the qubo section of `finance.ipynb` because we submit qubos to dwave and dwave support is moving from amazon braket to marketplace (which would cost us $70000 a year to access). If we decide to continue dwave support, this change can be reverted.

**Pull request merged**